app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,19 @@
+<h1><%= @recipe.name %></h1>
+<p>Complexity: <%= @recipe.complexity %></p>
+<p>Genre: <%= @recipe.genre %></p><br/>
+
+<% unless @recipe.ingredients.empty? %>
+  <h2>Ingredients:</h2>
+  <ul id="ingredients">
+    <% @recipe.ingredients.each do |ingredient| %>
+        <li><strong><%= ingredient.name %></strong>
+    <% end %>
+  </ul>
+<% end %><br/>
+
+<h3>Total Cost: $<%= @recipe.total_cost %></h3><br/>
+
+<form action="/recipes/<%= @recipe.id %>" method="post">
+  <label for="ingredient_id">Add an Ingredient to this Recipe by Ingredient ID</label>
+  <input type="number" name="ingredient_id" id="ingredient_id">
+  <input type="submit" value="Submit">
```

Comment:
Excellent work getting through the extensions. I would like to see you start getting more comfortable with Rails built in form helpers, but this absolutely works.